beamer/l1_resolution.py

```diff
@@ -9,25 +11,37 @@
 
 log = structlog.get_logger(__name__)
 
-_RELAYER_EXECUTABLE = "beamer-relayer"
+_RELAYER_NAMES = {"linux": "relayer-node16-linux-x64", "darwin": "relayer-node16-macos-x64"}
 
 
-def relayer_executable_exists() -> bool:
-    """Checks if the relayer executable is in PATH, return `False` if not"""
+def get_relayer_executable() -> Optional[Path]:
+    """Returns the path to the relayer executable if available, otherwise `None`"""
 
-    maybe_relayer = shutil.which(_RELAYER_EXECUTABLE)
-    if maybe_relayer is not None:
-        log.debug("Relayer found in PATH", relayer=maybe_relayer)
-        return True
+    bin_dir = Path(__file__).parent.joinpath("data/relayers/")
+    try:
+        maybe_relayer = bin_dir.joinpath(_RELAYER_NAMES[sys.platform])
+    except KeyError:
+        log.warning("Unsupported platform", platform=sys.platform)
+        return None
```

Comment:
Here I'd actually `raise RuntimeError("Unsupported platform ...` and let the agent automatically exit. Meaning we don't need to return `None`.

Reply:
Done.